packages/core/src/modules/dids/methods/peer/peerDidNumAlgo2.ts

```diff
@@ -115,20 +115,23 @@ export function didDocumentToNumAlgo2Did(didDocument: DidDocument) {
       typeof entry === 'string' ? didDocument.dereferenceVerificationMethod(entry) : entry
     )
 
-    // Transform als verification methods into a fingerprint (multibase, multicodec)
-    const encoded = dereferenced.map((entry) => {
+    // Transform all verification methods into a fingerprint (multibase, multicodec)
+    dereferenced.forEach((entry) => {
       const key = getKeyFromVerificationMethod(entry)
 
       // Encode as '.PurposeFingerprint'
       const encoded = `.${purpose}${key.fingerprint}`
 
-      return encoded
+      keys.push({ id: entry.id, encoded })
     })
-
-    // Add all encoded keys
-    did += encoded.join('')
   }
 
+  // Add all encoded keys ordered by their id (#key-1, #key-2, etc.)
+  did += keys
+    .sort((a, b) => a.id.localeCompare(b.id))
```

Comment:
I think this will fail in case of >= 10 keys within a did:peer, so we should use a regex/remove '#key-' prefix and do the sorting numerically

Reply:
Aren't the keys already sorted by default? I think we always process them start to end right?

Reply:
Given that our input is a modeled `DidDocument`, In `didDocumentToNumAlgo2Did()` we are sorting by key purpose (first assertionMethods, then keyAgreement, authentication, etc.), so it might happen that a key referenced by a service will not be encoded in the order we expect (e.g. mixing up verification key and key agreement key, or using key from another service). This is mainly the reason for this concern.